subsystems/intakesubsystem.py

```diff
@@ -0,0 +1,140 @@
+from enum import Enum, auto
+from commands2 import Subsystem
+from wpilib import RobotBase
+from wpimath.geometry import Rotation2d
+from ntcore import NetworkTableInstance
+from util.simtalon import Talon
+from util.simcoder import CTREEncoder
+from util.angleoptimize import intakeAccountForSillyEncoder
+
+import constants
+
+
+class IntakeSubsystem(Subsystem):
+    class IntakeState(Enum):
+        Intaking = auto()
+        Score = auto()
+        Scoring = auto()
+        Knock = auto()
+
+    def __init__(self) -> None:
+        Subsystem.__init__(self)
+        self.setName(__class__.__name__)  # basic subsystem boilerplate
+
+        self.pivotEncoder = CTREEncoder(
+            constants.kPivotEncoderID, constants.kPivotEncoderOffset
+        )
+
+        self.intakeMotor = Talon(
```

Comment:
each of these motors and the encoder needs to be added to the motor simulator in physics.py (see MotorSim)

Reply:
will be done on a new branch after elevator and intake are merged